src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -210,7 +220,24 @@ private void runRepair() {
         LOG.info("Repair for segment {} started, status wait will timeout in {} millis", segmentId,
             timeout);
         try {
-          condition.await(timeout, TimeUnit.MILLISECONDS);
+          long startTime = System.currentTimeMillis();
+          long maxTime = startTime + timeoutMillis;
+          
+          // If timeout is lower than 1mn, use timeout, otherwise we'll loop every minute to renew lead on segment
+          long waitTime = timeoutMillis<60000?timeoutMillis:60000;
+          
+          long lastLoopTime = System.currentTimeMillis();
+          while (System.currentTimeMillis() < maxTime) {
+            condition.await(waitTime, TimeUnit.MILLISECONDS);
+            if(lastLoopTime + 60_000 > System.currentTimeMillis() || context.storage.getRepairSegment(segmentId).get().getState() == RepairSegment.State.DONE){
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)